### PR TITLE
19 sponsors

### PR DESCRIPTION
**Pull request by steakunderscore**
This PR super seeds [PR#205](https://github.com/montrealrb/Montreal.rb/pull/205). Changed PR because the branched changed from my fork, to the Montreal.rb repo.

### CODE REVIEW

**Comment by steakunderscore** [comment]
The only thing that I still would like to do is to have the images sized a little better. But they always look a bit weird on this monitor, so I'm happy to leave it for now.

**Comment by mathieugagne** [comment]
Using cloudinary, you can request any size image you want on the spot. If we upload our images there, we could request images however we want them, padding, etc. included. Disclaimer: Haven't used it in a while but it could be a good fit for a small website like ours. Plus it's free.

**Comment by steakunderscore** [comment]
@mathieugagne - Yep, makes sense. There's already a [couple](https://github.com/montrealrb/Montreal.rb/pull/220/files#diff-714ba12e00360b67df9a26decec9bb00R8) of [methods](https://github.com/montrealrb/Montreal.rb/pull/220/files#diff-714ba12e00360b67df9a26decec9bb00R12) for this. But they need tweaking to make sure we aren't breaking companies logos. This patch can still be used as is.

File: app/models/organization.rb [hunk 1]
```diff
@@ -5,5 +5,8 @@ class Organization < ActiveRecord::Base
   translates :description
   mount_uploader :logo, LogoUploader
 
+  has_many   :sponsorships
+  has_many   :sponsors, through: :sponsorships, source: :event
```
Comment (sophiedeziel) [inline app/models/organization.rb]: Maybe `has_many :events, through: :sponsorships` ?

File: app/views/events/_event.html.erb [hunk 2]
```diff
@@ -16,5 +16,14 @@
           </div>
         <% end %>
     <% end %>
+    <% if event.sponsors.present? %>
+      <h3>Brought to you by our sponsors: </h3>
+      <% event.sponsors.each do |sponsor| %>
+        <h4><%= link_to sponsor.name, organization_path(sponsor) %></h4>
+        <% if sponsor.logo %>
+          <%= image_tag sponsor.logo.thumbnail.url %>
```
Comment (sophiedeziel) [inline app/views/events/_event.html.erb]: There's a `cl_image_tag` helper provided with the gem (very useful for "on-the-fly" formats)

```
<%= cl_image_tag sponsor.logo.thumbnail %>
```

File: db/schema.rb [hunk 3]
```diff
@@ -168,6 +168,16 @@
 
   add_index "pages", ["slug"], name: "index_pages_on_slug", unique: true, using: :btree
 
+  create_table "sponsorships", id: false, force: :cascade do |t|
+    t.integer  "event_id",        null: false
+    t.integer  "organization_id", null: false
```
Comment (sophiedeziel) [inline db/schema.rb]: Can the sponsorship have a "message" field? A text type would be great. That would be VERY useful to display a customized announcement for the sponsorship.

**Comment by sophiedeziel** [comment]
By the way, congratulations!! Hound didn't make any comment!!!!

**Comment by steakunderscore** [comment]
Thanks! :-D

File: app/models/organization.rb (hunk 1 above)
Comment (sophiedeziel) [inline app/models/organization.rb]: Or `has_many   :sponsored_events, through: :sponsorships, source: :event`
to be more precise

File: db/schema.rb (hunk 3 above)
Comment (steakunderscore) [inline db/schema.rb]: Sure thing.

**Comment by sophiedeziel** [comment]
I will review ASAP, thanks for the changes!

File: app/views/sponsorships/_sponsorship.html.erb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<div class="sponsorship">
+  <div class="row">
+    <div class="col-med-10 hidden-sm">
+      <h4>
+        <%= link_to sponsorship.organization.name, organization_path(sponsorship.organization) %>
+      </h4>
+      <% if sponsorship.message.present? %>
+        <p><%= sponsorship.message %></p>
+      <% end %>
+      </div>
```
Comment (sophiedeziel) [inline app/views/sponsorships/_sponsorship.html.erb]: Remove that extra closing `div`

Comment (steakunderscore) [inline app/views/sponsorships/_sponsorship.html.erb]: Oh man, good spotting! Will fix later tonight or tomorrow.

File: db/schema.rb [hunk 5]
```diff
@@ -168,6 +168,17 @@
 
   add_index "pages", ["slug"], name: "index_pages_on_slug", unique: true, using: :btree
 
+  create_table "sponsorships", force: :cascade do |t|
+    t.integer  "event_id",        null: false
```
Comment (sophiedeziel) [inline db/schema.rb]: The absence of `id` for sponsorship breaks the sponsorship dashboard... Can you add it?

File: app/views/sponsorships/_sponsorship.html.erb (hunk 4 above)
Comment (sophiedeziel) [inline app/views/sponsorships/_sponsorship.html.erb]: Took me a while to spot! 😄 It made the front-page sidebar appear at the bottom of the page

**Comment by sophiedeziel** [comment]
That div is the only missing thing! Ready to merge after that change

**Pull request merged**